### ISSUE TITLE
Spec Update: Specify how the stable URI works

### ISSUE THREAD

**Issue by jdesrosiers** [json-schema-org/json-schema-spec#1421]
The stable dialect URI identifies a schema as using the latest version of the standard dialect. What schema this URI identifies depends on what the implementation supports. Generally, the implementation will associate the stable URI with the meta-schema for the latest release they support. However, implementations may use a custom meta-schema that better represents what it supports. For example, if the implementation supports all of 2024 and some of 2025, they might want to use a custom meta-schema that includes only the 2025 keywords they support.

**Comment by gregsdennis** [comment]
> However, implementations may use a custom meta-schema that better represents what it supports. For example, if the implementation supports all of 2024 and some of 2025, they might want to use a custom meta-schema that includes only the 2025 keywords they support.

Are you thinking about things like VSCode which supports 2020-12 except for `$dynamic*` (and maybe a few other bits)?

Otherwise, I find it difficult to believe that an implementation will go to the effort of creating their own meta-schema to list partial support instead of using the latest meta-schema in their lib and documenting that they don't support some things.

I think the solution is to make this a harder requirement:

"Implementations MUST associate the stable URI with the meta-schema for the latest release they _fully_ support.  If an implementation supports individual features from one or more subsequent releases but does not fully support those releases, the implementation MUST create a custom meta-schema to identify those additional features it supports."

This way, it's clear what the stable URI means (in the context of that implementation).  If they want to _also_ support a subset of the next release(s), they are required to create a custom meta-schema.

I think this approach will better encourage full adoption of releases.

---

I expect we'll also end up with implementations, such as VSCode in its current state, where they say they support release _X_ except for features _a/b/c_ but they still associate the stable URI with release _X_ even though they don't have full support.  Maybe this is where Bowtie can help?

**Comment by gregsdennis** [comment]
> I think the solution is to make this a harder requirement

Basically, I feel that if we don't require something, implementations won't do it.

**Comment by gregsdennis** [comment]
> You still can't claim support for 2025 until you support all of it

This is what I think implementations will do anyway.  They'll claim support, then document that they don't support features _a/b/c_.

For example, despite [@Relequestual's bid](https://github.com/microsoft/vscode-json-languageservice/issues/116) to get correct messaging in the editor, VSCode still [documents](https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings) that it supports up through 2020-12 when it clearly does not.

> The JSON support shipped with VS Code supports all draft versions from Draft 4 to JSON Schema Draft 2020-12.

They don't even list the features they don't support!

**Comment by jdesrosiers** [comment]
> This is what I think implementations will do anyway. They'll claim support, then document that they don't support features a/b/c.

You're right. Some implementations are going to do whatever they want despite what we say or what the spec says. Making the spec stricter isn't going to change the behavior of people who already have no respect for what the spec says. They're going to continue to do whatever they want regardless. I think all we achieve by making things stricter is making things harder for those who do follow the rules.

**Comment by jdesrosiers** [comment]
I remembered another reason why allowing for 2024+ was important to me. It allows implementations to add support for new keywords as soon as they enter the stabilization process rather than having to wait until there's a release. People can start using them immediately and they can get usage and be stabilized sooner.

**Comment by gregsdennis** [comment]
> It allows implementations to add support for new keywords as soon as they enter the stabilization process rather than having to wait until there's a release.

Resolved here: https://github.com/orgs/json-schema-org/discussions/438

**Comment by gregsdennis** [comment]
Per https://github.com/orgs/json-schema-org/discussions/671#discussioncomment-8753266, we no longer have a single stable URI.

The meta-schema URI will work as follows:

- `https://json-schema.org/<version>` identifies the latest release within a specific version
- `https://json-schema.org/<version>/<year>` identifies a specific release

I'd still like to see some discussion particularly around the idea of dropping the URL URIs in favor of some other "unlocatable" scheme, but for now we can stick with `https`.

---

Action for this issue is to add these requirements to the specification, ideally by explicitly listing the version and year rather than the "template" shown above.
